system/jlib/jsocket.cpp

```diff
@@ -1547,6 +1562,14 @@ void CSocket::setTraceName()
 #endif
 }
 
+void CSocket::setPeerEndpoint(const SocketEndpoint &ep)
+{
+    assertex(!peerEpSet);
+    peerEpCrit.enter();
+    peerEpSet = true;
```

Comment:
This should be set after peerEP, otherwise theoretically peerEp can be read from getPeerEndpoint() before it is set.

Reply:
yes, changed.